storage/mysql/mysql_test.go

```diff
@@ -28,14 +63,253 @@ func TestMain(m *testing.M) {
 		}
 		klog.Fatalf("Failed to open MySQL test db: %v", err)
 	}
+	defer func() {
+		if err := db.Close(); err != nil {
+			klog.Warningf("Failed to close MySQL database: %v", err)
+		}
+	}()
 	if err := db.PingContext(ctx); err != nil {
 		if *isMySQLTestOptional {
 			klog.Warning("MySQL not available, skipping all MySQL storage tests")
 			return
 		}
 		klog.Fatalf("Failed to ping MySQL test db: %v", err)
 	}
+	testDB = db
+
 	klog.Info("Successfully connected to MySQL test database")
 
+	initDatabaseSchema(ctx)
+
+	noteSigner, err = note.NewSigner(testPrivateKey)
+	if err != nil {
+		klog.Fatalf("Failed to create new signer: %v", err)
+	}
+	noteVerifier, err = note.NewVerifier(testPublicKey)
+	if err != nil {
+		klog.Fatalf("Failed to create new verifier: %v", err)
+	}
+
 	os.Exit(m.Run())
 }
+
+// initDatabaseSchema drops the tables and then imports the schema.
+func initDatabaseSchema(ctx context.Context) {
+	dropTablesSQL := "DROP TABLE IF EXISTS `Checkpoint`, `Subtree`, `TiledLeaves`"
+
+	rawSchema, err := os.ReadFile("schema.sql")
+	if err != nil {
+		klog.Fatalf("Failed to read schema.sql: %v", err)
+	}
+
+	db, err := sql.Open("mysql", *mysqlURI+"?multiStatements=true")
+	if err != nil {
+		klog.Fatalf("Failed to connect to DB: %v", err)
+	}
+	defer func() {
+		if err := db.Close(); err != nil {
+			klog.Warningf("Failed to close db: %v", err)
+		}
+	}()
+
+	if _, err := db.ExecContext(ctx, dropTablesSQL); err != nil {
+		klog.Fatalf("Failed to drop all tables: %v", err)
+	}
+
+	if _, err := db.ExecContext(ctx, string(rawSchema)); err != nil {
+		klog.Fatalf("Failed to execute init database schema: %v", err)
+	}
+}
+
+func TestNew(t *testing.T) {
+	ctx := context.Background()
+
+	for _, test := range []struct {
+		name    string
+		opts    []func(*tessera.StorageOptions)
+		wantErr bool
+	}{
+		{
+			name:    "no tessera.StorageOption",
+			opts:    nil,
+			wantErr: true,
+		},
+		{
+			name: "standard tessera.WithCheckpointSignerVerifier",
+			opts: []func(*tessera.StorageOptions){
+				tessera.WithCheckpointSignerVerifier(noteSigner, noteVerifier),
+			},
+			wantErr: false,
```

Comment:
nit: can just leave these out and rely on the default value of bool being `false`.

Reply:
Done.